Wizard pages should be able to specify their minimum size.

ptziegler opened the pull request:
Subclasses of WizardPage can overwrite the getMinimumPageSize() method in order to set a minimal size for the wizard shell. By default, no such minimum size is set.

For the sake of backwards-compatibility, I've moved this logic into "supplement" interfaces, similar to how it's done for the IWizardContainer. Because there are already several classes implementing IWizardPage directly and would otherwise have to implement a dummy method.

If getMinimumPageSize() is not overwritten, then the minimum size of the shell is not set. This corresponds to the current behavior. Otherwise the minimum size is of the shell is calculated to be equal or larger than the minimum sizes of all wizard pages.

Resolves #516

vogella commented:
I suggest to extend the original interface with a an empty default implementation instead of creating new interfaces.

ptziegler committed:
Wizard pages should be able to specify their minimum size.

Adds the new methods getMinimumWizardSize() to the IWizard and
getMinimumPageSize() to the IWizardPage interface. It is currently not
possible to control the minimum size of the wizard page, if the wizard
is not created by the application. You can take the target platform
editor as an example, where custom wizards may be contributed via an
extension point.

Subclasses of WizardPage can overwrite the getMinimumPageSize() method
in order to set a minimal size for the wizard shell. By default, no such
minimum size is set.

If getMinimumPageSize() is not overwritten, then the minimum size of the
shell is not set. This corresponds to the current behavior. Otherwise
the minimum size is of the shell is calculated to be equal or larger
than the minimum sizes of all wizard pages.

ptziegler commented:
> I suggest to extend the original interface with a an empty default implementation

That's a good point. I wasn't exactly sure what the convention is when dealing with API extensions, so I took the IWizardContainer as reference.

I assume this "pattern" is a relic from the pre-Java 8 era, before default methods were supported. So this was the only way to implement such extensions without breaking backwards compatibility.

jukzi commented:
@vogella is there anything missing or submit?

vogella commented:
Thanks @ptziegler 

Also thanks to @jukzi for the reminder.
